Documentation/scenario-test-guide.md

```diff
@@ -0,0 +1,51 @@
+
+Regardless which platform where you run WCF client tests, you will need a WCF test service for scenario tests to work. WCF Test services can be hosted in two ways like WCF users would usually do, ie., self-hosted and IIS-hosted. The following sections provide instructions of how to host WCF test services and how to run WCF scenario tests.
+
+Self-Hosted WCF Test Service
+----------------------------
+To start a self-hosted WCF test service on a Windows machine, execute this script that is located at directory `src\System.Private.ServiceModel\tools\scripts`.
+```
+    StartWCFSelfHostedSvc.cmd
+```
+This script will add necessary firewall rules, generate and install self-signed test certificates needed by tests, build and kick off a self-hosted WCF test service process from project at `src\System.Private.ServiceModel\tools\SelfHostedWcfService`. When you are done with this WCF test service, type `exit` and press `enter` in the window where the service was started. Upon shutting down the service, this script will also clean up firewall rules and remove test certificates it installed.
+
+This script requires elevated privileges. If it is run from a non-elevated window, you may be prompted for elevated permission.
+
+IIS-Hosted WCF Test Service
+---------------------------
+To host WCF services on IIS, the Windows machine has these prerequisites
+
+1. Web Server (IIS) and Application Server roles are installed. This includes ASP.NET and WebSocket protocol.
+2. .NET Framework 4.x features including WCF service and activation features are enabled.
+3. Ports 80 (HTTP), 443 (HTTPS) and 808 (TCP) are opened.
+
+Copy script `src\System.Private.ServiceModel\tools\scripts\SetupWcfIISHostedService.cmd` to the Windows machine and run it in an elevated command window.
+```
+    SetupWcfIISHostedService.cmd {id}
+```
+
+where `id` is an identifier that will be appended to the name of the IIS application to be created, for example, command `SetupWcfIISHostedService.cmd Test` will create an IIS aplication named "WCFService*Test*". This script will create a clone of WCF GitHub repository, an IIS application pool and an IIS application from `src\System.Private.ServiceModel\tools\IISHostedWcfService`. The script will also install necessary test certificates and add IIS HTTPS binding for testing.
+
+This is one time setup. Once you set it up, you do not need to build or start the service manually. IIS will build and activate the service automatically upon the first request from tests. 
+
+Runing WCF Scenario Tests
+-------------------------
+As instructed in [Developer Guide](developer-guide.md#running-scenario-tests), to run scenario tests, you will need to specify the WCF service base address via `/p:ServiceUri` parameter to the build script.
```

Comment:
My preference would be to make this scenario test guide *not* refer back to the developer guide but to keep all information about running scenario tests just in this scenario guide, and then to link to it from the developer guide.  I don't feel strongly either way so you choose which seems better.